ecs-cli/modules/clients/aws/secretsmanager/client.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2015-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package secretsmanager
+
+import (
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/clients"
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/config"
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/aws/aws-sdk-go/service/secretsmanager/secretsmanageriface"
+)
+
+// SMClient defines methods for interacting with the SecretsManagerAPI interface
+type SMClient interface {
+	CreateSecret(secretsmanager.CreateSecretInput) (*secretsmanager.CreateSecretOutput, error)
+	ListSecrets(*string) (*secretsmanager.ListSecretsOutput, error)
+}
+
+type secretsManagerClient struct {
+	client secretsmanageriface.SecretsManagerAPI
+	config *config.CommandConfig
```

Comment:
It doesn't look like the config field is used in any of the methods- what's the intention behind adding it here?

Reply:
config should be used for the client session; see comment above about using sdk factory method.

Reply:
This is another thing that seems to be common in some of our existing clients but whose use is not clear to me. If the only thing we need it for is the session and we don't need to mock that in tests, then it looks like we can omit it from here.

Reply:
Since none of the code ends up using this field, I believe it should be removed. Per our discussion in person yesterday, it is true that the CloudFormation client includes the config as a field, but its also not used there, and looks like there probably wasn't much thought put into it.

Reply:
Agreed, I'm going to rip it out :P